### PR TITLE
[OPENJDK-2199] Update tests to refer new application source repository

### PR DESCRIPTION
**Pull request by jhuttana**
This is in support of https://github.com/jboss-container-images/openjdk-test-applications/pull/1 which in-turn addresses
Jira ticket: https://issues.redhat.com/browse/OPENJDK-2199
GH Issue: https://github.com/jboss-container-images/openjdk/issues/400

Could you please take a look?

### CODE REVIEW

File: tests/features/java.security.feature [hunk 1]
```diff
@@ -2,5 +2,5 @@
 @ubi9/openjdk-17
 Feature: Openshift S2I tests
   Scenario: Check networkaddress.cache.negative.ttl has been set correctly
-    Given s2i build https://github.com/jboss-openshift/openshift-examples/ from binary-cli-security-property
+    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from binary-cli-security-property
```
Comment (jmtd) [inline tests/features/java.security.feature]: This URI is not a valid Git repository. It results in the following error:
```
ERROR:cekit:E1024 13:55:37.768913  155305 git.go:188] Clone failed: source https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources#master, target /tmp/s2i648681403/upload/tmp, with output "fatal: repository 'https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources/' not found\n"
```

In order to make the behave tests reference a different git branch, you need to change the "Given" line, in a way like this:

```diff
-    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from undertow-servlet
+    Given s2i build https://github.com/jhuttana/openjdk-test-applications from undertow-servlet using pick_relevant_sources
```

The list of valid "Given" commands is controlled by the "behave-test-steps" repository. For example https://github.com/cekit/behave-test-steps/blob/v1/steps/s2i_steps.py#L53

For most of the "Given" command that we currently use, there should be an alternative that lets you specify the branch (confusingly phrased as "tag" in the behave-test-steps source.) There might be some tests where there is not an appropriate "GIven" command already: in which case, we can add one to behave-test-steps.

Comment (jmtd) [reply tests/features/java.security.feature]: Alternatively, if you configure your GitHub fork of openjdk-test-applications to make the branch "pick_relevant_sources" the default, you would not need to specify it in these tests.

Comment (jhuttana) [reply tests/features/java.security.feature]: > This URI is not a valid Git repository. It results in the following error:
> 
> ```
> ERROR:cekit:E1024 13:55:37.768913  155305 git.go:188] Clone failed: source https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources#master, target /tmp/s2i648681403/upload/tmp, with output "fatal: repository 'https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources/' not found\n"
> ```
> 
> In order to make the behave tests reference a different git branch, you need to change the "Given" line, in a way like this:
> 
> ```diff
> -    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from undertow-servlet
> +    Given s2i build https://github.com/jhuttana/openjdk-test-applications from undertow-servlet using pick_relevant_sources
> ```
> 
> The list of valid "Given" commands is controlled by the "behave-test-steps" repository. For example https://github.com/cekit/behave-test-steps/blob/v1/steps/s2i_steps.py#L53
> 
> For most of the "Given" command that we currently use, there should be an alternative that lets you specify the branch (confusingly phrased as "tag" in the behave-test-steps source.) There might be some tests where there is not an appropriate "GIven" command already: in which case, we can add one to behave-test-steps.

I have updated the URI in the "Given" section.

Comment (jhuttana) [reply tests/features/java.security.feature]: > Alternatively, if you configure your GitHub fork of openjdk-test-applications to make the branch "pick_relevant_sources" the default, you would not need to specify it in these tests.

Thanks for letting me know about this :)

**Comment by jmtd** [comment]
Thank you for doing that. By referencing your fork of the new openjdk-test-applications repository, and that specific branch, you have proven that all the relevant quickstarts needed by the test suite have been correctly identified and copied into that repository. I have just now merged https://github.com/jboss-container-images/openjdk-test-applications/pull/1 into openjdk-test-applications.

Since that is now merged, the tests *here* need updating to refer to that repository, rather than you fork. That means changing them all again in this PR (sorry!), to something like

```
+    Given s2i build https://github.com/jboss-container-images/openjdk-test-applications from undertow-servlet
```

(jboss-container-images URI, no explicit "using branchname" suffix)

**Comment by jhuttana** [comment]
> Thank you for doing that. By referencing your fork of the new openjdk-test-applications repository, and that specific branch, you have proven that all the relevant quickstarts needed by the test suite have been correctly identified and copied into that repository. I have just now merged [jboss-container-images/openjdk-test-applications#1](https://github.com/jboss-container-images/openjdk-test-applications/pull/1) into openjdk-test-applications.
> 
> Since that is now merged, the tests _here_ need updating to refer to that repository, rather than you fork. That means changing them all again in this PR (sorry!), to something like
> 
> ```
> +    Given s2i build https://github.com/jboss-container-images/openjdk-test-applications from undertow-servlet
> ```
> 
> (jboss-container-images URI, no explicit "using branchname" suffix)

When I tried to change the URI in the Given section like 
```
+    Given s2i build https://github.com/jboss-container-images/openjdk-test-applications from undertow-servlet
```
The test was failing. Then I tried something like
```
+    Given s2i build https://github.com/jboss-container-images/openjdk-test-applications from undertow-servlet using main
```
Then the test passed. That led me to check for the default branch name that is used in the old application repo https://github.com/jboss-openshift/openshift-examples/ and that is `master`. Also the failing scenario had a line in its trace
```
2023-11-08 01:07:17,798 steps.py:59         ERROR I1108 01:07:17.798651  623008 scm.go:20] DownloadForSource https://github.com/jboss-container-images/openjdk-test-applications#master
2023-11-08 01:07:20,515 steps.py:59         ERROR I1108 01:07:17.798664  623008 sti.go:206] Preparing to build integ-ubi9/openjdk-17
2023-11-08 01:07:20,515 steps.py:59         ERROR I1108 01:07:17.798732  623008 clone.go:35] Downloading "https://github.com/jboss-container-images/openjdk-test-applications#master" ("undertow-servlet") ...
```
where is it trying to download from `master` branch.
So we have two options to fix this
1. In the new application source repository change the default branch name from `main` to  `master` 
Or
2. Configure Github to reference the default branch as `main`.
As I don't have required permissions to do any of these changes could you please help?

**Comment by jmtd** [comment]
Good catch @jhuttana -- the root issue is, I think, behave-test-steps defaulting to "master" even when it doesn't exist. I've filed https://github.com/cekit/behave-test-steps/issues/49 , and renamed "main" to "master" in https://github.com/jboss-container-images/openjdk-test-applications for the time being.

**Comment by jhuttana** [comment]
> Good catch @jhuttana -- the root issue is, I think, behave-test-steps defaulting to "master" even when it doesn't exist. I've filed [cekit/behave-test-steps#49](https://github.com/cekit/behave-test-steps/issues/49) , and renamed "main" to "master" in https://github.com/jboss-container-images/openjdk-test-applications for the time being.

Thank you Jon :) I will make the suitable changes to tests and try to test locally.

**Comment by jhuttana** [comment]
All the tests in this PR are now updated to refer to master application source repo.
Then tested with @wip tag
```
$ cekit --descriptor ubi9-openjdk-17.yaml test behave --wip
.
.
.
8 features passed, 0 failed, 4 skipped
25 scenarios passed, 0 failed, 34 skipped
87 steps passed, 0 failed, 83 skipped, 0 undefined
Took 45m50.057s
```

**Pull request merged**